scrapy_autoextract/cache.py

```diff
@@ -39,8 +42,30 @@ def __str__(self):
 
 
 class AutoExtractCache(_Cache):
-    def __init__(self, path):
-        self.db = sqlitedict.SqliteDict(path, autocommit=True)
+    def __init__(self, path, *, compressed=True):
+        self.compressed = compressed
+        tablename = 'responses_gzip' if compressed else 'responses'
+        self.db = sqlitedict.SqliteDict(path,
+                                        tablename=tablename,
+                                        autocommit=True,
+                                        encode=self.encode,
+                                        decode=self.decode)
+
+    def encode(self, obj):
+        # based on sqlitedict.encode
+        data = pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)
+        if self.compressed:
+            data = gzip.compress(data, compresslevel=3)
```

Comment:
TIL gzip level 3 is almost twice as fast as level 6 and its compression power is still good.

https://www.rootusers.com/gzip-vs-bzip2-vs-xz-performance-comparison/